tests/jest/unit/modules/kit-elements-defaults/assets/js/editor/store.test.js

```diff
@@ -0,0 +1,152 @@
+import store from 'elementor/modules/kit-elements-defaults/assets/js/editor/store';
+import { MockFetch } from './utils';
+
+const wpApiSettings = {
+	root: 'http://example.com/wp-json/',
+	nonce: '123456',
+};
+
+describe( 'modules/kit-elements-defaults/assets/js/editor/store.js', () => {
+	let fock, originalFetch;
+
+	beforeEach( () => {
+		window.wpApiSettings = wpApiSettings;
+
+		fock = new MockFetch( wpApiSettings.root + 'elementor/v1' );
+
+		originalFetch = window.fetch;
+
+		window.fetch = fock.getFetchMock();
+	} );
+
+	afterEach( () => {
+		window.fetch = originalFetch;
+
+		delete window.wpApiSettings;
+	} );
+
+	it( 'Should return empty object for non-existing type', () => {
+		// Act & Assert.
+		expect( store.get( 'non-existing-element' ) ).toStrictEqual( {} );
+	} );
+
+	it( 'Should load items to cache', async () => {
+		// Arrange.
+		fock.get( '/kit-elements-defaults' )
+			.reply( 200, {
+				section: {
+					color: 'red',
+					background_color: '#F00',
+				},
+				button: {
+					type: 'info',
+				},
+			} );
+
+		// Act.
+		await store.load();
+
+		// Assert.
+		const section = store.get( 'section' );
+		const button = store.get( 'button' );
+
+		expect( section ).toStrictEqual( {
+			color: 'red',
+			background_color: '#F00',
+		} );
+
+		expect( button ).toStrictEqual( {
+			type: 'info',
+		} );
+	} );
+
+	it( 'Should upsert and refresh cache', async () => {
+		// Arrange.
+		store.items = {
+			section: {
+				old_control: 'old_value',
+			},
+		};
+
+		fock.put( '/kit-elements-defaults/section' )
+			.reply( 201, null );
+
+		fock.get( '/kit-elements-defaults' )
+			.reply( 200, {
+				section: {
+					new_control: 'new_value',
+				},
+			} );
+
+		// Act.
+		await store.upsert( 'section', {
+			new_control: 'new_value',
+		} );
+
+		// Assert.
+		const section = store.get( 'section' );
+
+		expect( section ).toStrictEqual( {
+			new_control: 'new_value',
+		} );
+	} );
+
+	it( 'Should delete and refresh cache', async () => {
+		// Arrange.
+		store.items = {
+			section: {
+				old_control: 'old_value',
+			},
+		};
+
+		fock.delete( '/kit-elements-defaults/section' )
+			.reply( 204, '' );
+
+		fock.get( '/kit-elements-defaults' )
+			.reply( 200, {} );
+
+		// Act.
+		await store.delete( 'section' );
+
+		// Assert.
+		const section = store.get( 'section' );
+
+		expect( section ).toStrictEqual( {} );
+	} );
+
+	it( 'Should send proper auth headers', async () => {
```

Comment:
It seems like this test case testing the implementation details, maybe it can be removed?

Reply:
Yeah, I wasn't sure about this one. On the one hand, it tests the implementation. On the other hand, it's kinda an API that we expose and we need to make sure that it communicates properly with the server. and since we don't have a real server, we can't really test it without knowing  the implementation. WDYT?